gnuplot/detailed.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Plot all benchmarks as separate SVGs.
+
+# Make sure gnuplot is installed.
+if ! [ -x "$(command -v gnuplot)" ]; then
+    echo "command not found: gnuplot"
+    exit 1
+fi
+
+# Ensure at least one input and output file is present.
+if [ $# -lt 2 ]; then
+    echo "Usage: gnuplot.sh <INPUT_FILES>... <OUTPUT_DIRECTORY>"
+    exit 1
+fi
+
+# Get last argument as output directory.
+output_index=$#
+output_directory=${!output_index%/}
+mkdir -p "$output_directory"
+
+num_cols=$(cat "$1" | head -n 1 | awk '{ print NF }')
+
+for col in $(seq 1 $num_cols); do
+    # Append benchmark name before file suffix.
+    benchmark=$(cat "$1" | head -n 1 | awk "{ print \$$col }")
```

Comment:
I think you can use `cut -d ' ' -f $col` instead of awk?

Reply:
But why? That seems significantly more complicated.

Reply:
it's not `cut` is a right tool for such things.